Flaky test: test_backup_restore_on_cluster/test_cancel_backup.py::test_short_disconnection_doesnt_stop_backup

_Important: This issue was automatically generated and is used by CI for matching failures. DO NOT modify the body content. DO NOT remove labels._

Test name: test_backup_restore_on_cluster/test_cancel_backup.py::test_short_disconnection_doesnt_stop_backup
Failure reason: AssertionError
CI report: [Integration tests (arm_binary, distributed plan, 4/4)](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=98189&sha=a7206992b7b810cb6a3e18f236fc4f67bf9f6fbf&name_0=PR&name_1=Integration%20tests%20%28arm_binary%2C%20distributed%20plan%2C%204%2F4%29)

Failing test history: [cidb](https://play.clickhouse.com#V0lUSAogICAgOTAgQVMgaW50ZXJ2YWxfZGF5cwpTRUxFQ1QKICAgIHRvU3RhcnRPZkRheShjaGVja19zdGFydF90aW1lKSBBUyBkYXksCiAgICBjb3VudCgpIEFTIGZhaWx1cmVzLAogICAgZ3JvdXBVbmlxQXJyYXkocHVsbF9yZXF1ZXN0X251bWJlcikgQVMgcHJzLAogICAgYW55KHJlcG9ydF91cmwpIEFTIHJlcG9ydF91cmwKRlJPTSBjaGVja3MKV0hFUkUgKG5vdygpIC0gdG9JbnRlcnZhbERheShpbnRlcnZhbF9kYXlzKSkgPD0gY2hlY2tfc3RhcnRfdGltZQogICAgQU5EIHRlc3RfbmFtZSA9ICd0ZXN0X2JhY2t1cF9yZXN0b3JlX29uX2NsdXN0ZXIvdGVzdF9jYW5jZWxfYmFja3VwLnB5Ojp0ZXN0X3Nob3J0X2Rpc2Nvbm5lY3Rpb25fZG9lc250X3N0b3BfYmFja3VwJwogICAgLS0gQU5EIGNoZWNrX25hbWUgPSAnSW50ZWdyYXRpb24gdGVzdHMgKGFybV9iaW5hcnksIGRpc3RyaWJ1dGVkIHBsYW4sIDQvNCknCiAgICBBTkQgdGVzdF9zdGF0dXMgSU4gKCdGQUlMJywgJ0VSUk9SJykKICAgIEFORCAocHVsbF9yZXF1ZXN0X251bWJlciA9IDAgT1IgYmFzZV9yZWYgSU4gKCdtYXN0ZXInKSkKICAgIEFORCB0ZXN0X2NvbnRleHRfcmF3IExJS0UgJyVBc3NlcnRpb25FcnJvciUnCkdST1VQIEJZIGRheQpPUkRFUiBCWSBkYXkgREVTQwo=)

Test output:
```
File: test_backup_restore_on_cluster/test_cancel_backup.py:661 - in test_short_disconnection_doesnt_stop_backup
    with NoTrashChecker() as no_trash_checker, ConfigManager() as config_manager:
File: test_backup_restore_on_cluster/test_cancel_backup.py:375 - in __exit__
    assert (error in self.expect_errors) or (error in self.allow_errors)
E   AssertionError: assert ('FAILED_TO_SYNC_BACKUP_OR_RESTORE' in [] or 'FAILED_TO_SYNC_BACKUP_OR_RESTORE' in ['KEEPER_EXCEPTION', 'SOCKET_TIMEOUT', 'CANNOT_READ_ALL_DATA', 'NETWORK_ERROR', 'TABLE_IS_READ_ONLY', 'NO_REPLICA_HAS_PART'])
E    +  where [] = <test_backup_restore_on_cluster.test_cancel_backup.NoTrashChecker object at 0xff6028bae290>.expect_errors
E    +  and   ['KEEPER_EXCEPTION', 'SOCKET_TIMEOUT', 'CANNOT_READ_ALL_DATA', 'NETWORK_ERROR', 'TABLE_IS_READ_ONLY', 'NO_REPLICA_HAS_PART'] = <test_backup_restore_on_cluster.test_cancel_backup.NoTrashChecker object at 0xff6028bae290>.allow_errors
```

--- a/tests/integration/test_backup_restore_on_cluster/test_cancel_backup.py
+++ b/tests/integration/test_backup_restore_on_cluster/test_cancel_backup.py
@@ -666,6 +666,18 @@ def test_short_disconnection_doesnt_stop_backup():
             break
         time.sleep(1)
 
+    backup_process_counts = {
+        get_node_name(node): int(
+            node.query("SELECT count() FROM system.processes WHERE query_kind = 'Backup'")
+        )
+        for node in nodes
+    }
+    total_backup_processes = sum(backup_process_counts.values())
+    assert total_backup_processes == 0, (
+        "Backup queries still running after pre-test wait in "
+        "test_short_disconnection_doesnt_stop_backup: "
+        + ", ".join(f"{name}={count}" for name, count in backup_process_counts.items())
+    )
     create_and_fill_table(random_node())
 
     with NoTrashChecker() as no_trash_checker, ConfigManager() as config_manager:

--- a/tests/integration/test_backup_restore_on_cluster/test_cancel_backup.py
+++ b/tests/integration/test_backup_restore_on_cluster/test_cancel_backup.py
@@ -656,6 +656,16 @@ def test_long_disconnection_stops_backup():
 
 # A backup must NOT be stopped if Zookeeper is disconnected shorter than `failure_after_host_disconnected_for_seconds`.
 def test_short_disconnection_doesnt_stop_backup():
+    # Wait for any backup processes from the previous test to finish their ZooKeeper cleanup
+    # before creating the NoTrashChecker, so residual errors don't fall in its time window.
+    for _ in range(30):
+        if not any(
+            int(node.query("SELECT count() FROM system.processes WHERE query_kind = 'Backup'")) > 0
+            for node in nodes
+        ):
+            break
+        time.sleep(1)
+
     create_and_fill_table(random_node())
 
     with NoTrashChecker() as no_trash_checker, ConfigManager() as config_manager:
@@ -711,6 +721,16 @@ def test_short_disconnection_doesnt_stop_backup():
 
 # A restore must NOT be stopped if Zookeeper is disconnected shorter than `failure_after_host_disconnected_for_seconds`.
 def test_short_disconnection_doesnt_stop_restore():
+    # Wait for any backup processes from the previous test to finish their ZooKeeper cleanup
+    # before creating the NoTrashChecker, so residual errors don't fall in its time window.
+    for _ in range(30):
+        if not any(
+            int(node.query("SELECT count() FROM system.processes WHERE query_kind = 'Backup'")) > 0
+            for node in nodes
+        ):
+            break
+        time.sleep(1)
+
     # Make a backup.
     backup_id = get_backup_id_of_successful_backup()
 

--- a/tests/integration/test_backup_restore_on_cluster/test_cancel_backup.py
+++ b/tests/integration/test_backup_restore_on_cluster/test_cancel_backup.py
@@ -678,6 +678,7 @@ def test_short_disconnection_doesnt_stop_backup():
         "test_short_disconnection_doesnt_stop_backup: "
         + ", ".join(f"{name}={count}" for name, count in backup_process_counts.items())
     )
+
     create_and_fill_table(random_node())
 
     with NoTrashChecker() as no_trash_checker, ConfigManager() as config_manager:
@@ -742,6 +743,19 @@ def test_short_disconnection_doesnt_stop_restore():
         ):
             break
         time.sleep(1)
+
+    backup_process_counts = {
+        get_node_name(node): int(
+            node.query("SELECT count() FROM system.processes WHERE query_kind = 'Backup'")
+        )
+        for node in nodes
+    }
+    total_backup_processes = sum(backup_process_counts.values())
+    assert total_backup_processes == 0, (
+        "Backup queries still running after pre-test wait in "
+        "test_short_disconnection_doesnt_stop_backup: "
+        + ", ".join(f"{name}={count}" for name, count in backup_process_counts.items())
+    )
 
     # Make a backup.
     backup_id = get_backup_id_of_successful_backup()

